src/serial.rs

```diff
@@ -188,7 +319,7 @@ where
     pub fn new(
         usart: Usart,
         pins: (Tx, Rx),
-        baud_rate: Baud,
+        config: impl Into<config::Config>,
```

Comment:
I've seen it, felt in love and "pirated" it from https://github.com/stm32-rs/stm32h7xx-hal/blob/acd47beb4b84b4dc46da3a8b68688bc8c5984604/src/serial.rs#L421. All glory to @richardeoin! :)

> I wouldn't call this backwards compatible and still would raise the minor version to 0.8.0 (not that this does really matter), but this would allow a pretty smooth transition.

Sure thing. May be i should have put it in quote marks. I like this approach because it eliminates the need for changing the existing code in many situations I bet.

> Styling wise, I would not mix `impl ...` and where clauses maybe? Could you move this constraint to the where clauses?

Do you mean as a where clause to this function? Will do. That's pretty new terrain for me, so I will have a look how to do this and might come back with a question to you.

Reply:
I've meant this:

```diff
    pub fn new(
         usart: Usart,
         pins: (Tx, Rx),
-        config: impl Into<config::Config>,
+        config: Config,
         clocks: Clocks,
         apb: &mut <Usart as Instance>::APB,
         apb: &mut <Usart as Instance>::APB,
     ) -> Self
     where
         Usart: Instance,
         Tx: TxPin<Usart>,
         Rx: RxPin<Usart>,
+        Clocks: Into<config::Config>,
     {
```

In the end `var: impl ...` is just a shorthand for the where clause, 
but as where clauses are already used in this case, I would suggest to it 
for the config part as well.

Reply:
Thank you for guiding me here! Sure thing. In this case the where clause should be used consistently.

Reply:
Done. This was pretty interesting! Thank you! :)